examples/opf/tools/sp_plotter.py

```diff
@@ -260,10 +259,11 @@ def testSP():
       for i in xrange(inputSize):
         time.sleep(0.001)
         if iter == numIter - 1:
-          outputs[i] = sp.compute(inputs[i], learn=doLearn, infer=False)
+          sp.compute(inputs[i], learn=doLearn, activeArray=outputs[i])
```

Comment:
Agreed. I also think the first approach is better: it is more descriptive and explicit using a method return then simply passing reference parameters. The problem in the second approach is that you have to check inner and outer scope to figure out *which* value should be returned.